src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -392,7 +392,7 @@ private ArmClientExtensions EnsureArmClientExtensions()
         }
 
         private IEnumerable<ResourceData>? _resourceDatas;
-        public IEnumerable<ResourceData> ResourceData => _resourceDatas ??= RawRequestPathToResourceData.Values.Distinct();
+        public IEnumerable<ResourceData> ResourceData => _resourceDatas ??= RawRequestPathToResourceData.Values.Where(data => data is not VirtualResourceData).Distinct();
```

Comment:
Maybe rename this property to `NonVirtualResourceData`?

Reply:
I am struggling renaming this, or leave this as it is, and change the output part in `MgmtTarget` only to write a file out when it is not `VirtualResourceData`.
What do you think?